pandas/core/indexes/multi.py

```diff
@@ -2163,11 +2164,41 @@ def append(self, other):
             (isinstance(o, MultiIndex) and o.nlevels >= self.nlevels) for o in other
         ):
             arrays = []
-            for i in range(self.nlevels):
-                label = self._get_level_values(i)
-                appended = [o._get_level_values(i) for o in other]
-                arrays.append(label.append(appended))
-            return MultiIndex.from_arrays(arrays, names=self.names)
+            if self.names.count(None) > 1 or any(
+                o.names.count(None) > 1 for o in other
+            ):
+
+                for i in range(self.nlevels):
+
+                    label = self._get_level_values(i)
+                    appended = [o._get_level_values(i) for o in other]
+                    arrays.append(label.append(appended))
+                    index_label_list = self.names
+
+            else:
+                index_label_list = self.get_unique_indexes(other)
+
+                for index_label in index_label_list:
+
+                    index = self.get_index_data(
+                        data_index=self, column_name=index_label, other=other
+                    )
+                    appended = []
+
```

Comment:
Obtain the index data from the self dataframe.

Reply:
Search self is false here as the "self" Dataframe is searched being searched through here.